server/athenian/api/auth.py

```diff
@@ -273,4 +275,11 @@ def _get_token_auth_header(self, request: aiohttp.web.Request) -> str:
         except Exception:
             return aiohttp.web.Response(body="Your auth token is likely revoked.", status=401)
 
+        god = await request.sdb.fetch_one(select([God.mapped_id]).where(God.user_id == user.id))
+        if god is not None:
+            mapped_id = god[God.mapped_id.key]
+            if mapped_id is not None:
+                user.god_id = user.id
```

Comment:
🤔I don't understand how this works. I mean, shouldn't this be done independently from whether the god is currently mapped already? Otherwise [this](https://github.com/athenianco/athenian-api/pull/70/commits/c8484d150ba50ad54ad7f2d803b5fc60428440ce?file-filters%5B%5D=.py#diff-7640db81d011b0cff79abcd6886d5be6R40) would be always `None`, doesn't it?

Reply:
This should work like this. Let's suppose there is a super admin `vadim@athenian.co` and a regular user `marvin@athenian.co`

1. `vadim@athenian.co` logs in as usual.
2. Call `/v1/become/auth0|whatever-belongs-to-marvin@athenian.co`
3. A new record in the DB appears that maps `vadim@athenian.co` (`God.user_id`) to `marvin@athenian.co` (`God.mapped_id`).
4. Any subsequent request from `vadim@athenian.co` is first handled as normal, so Auth0 checks whether the user is `vadim@athenian.co`.
5. However, in the end, we check whether `vadim@athenian.co` is a god. If he is, we look up the mapped ID in the DB.
6. We query the mgmt Auth0 API to fetch the full profile of the mapped user - `marvin@athenian.co`.
7. We overwrite the `user` field of `request` and additionally set the extra attribute `god_id` to indicate that the user is a mapped god.
8. API handlers think that the user is `marvin@athenian.co`.
9. But `/v1/become` checks `user.god_id` and if it exists, it is used in the DB god check instead of the regular `user.id`. Thus we don't lose the ability to turn into any other user, including the empty string (`None`, the initial default unmapped state).

Reply:
Got it! Thanks for the detailed explanation!

Reply:
I should include it in the DD once merged.